source/uwp/Renderer/lib/XamlHelpers.cpp

```diff
@@ -668,538 +437,377 @@ namespace AdaptiveCards::Rendering::Uwp::XamlHelpers
-            XamlHelpers::AppendXamlElementToPanel(localInputLayout.Get(), stackPanelAsPanel.Get());
+            actualUIElement = inputLayout;
+            XamlHelpers::AppendXamlElementToPanel(inputLayout, inputStackPanel);
         }
 
         // Add the error message if there's validation and one exists
-        ComPtr<IUIElement> errorMessageControl;
-        XamlHelpers::RenderInputErrorMessage(adaptiveInput, renderContext, &errorMessageControl);
-        if (errorMessageControl != nullptr)
+        auto errorMessageControl = XamlHelpers::RenderInputErrorMessage(adaptiveInput, renderContext);
+
+        if (errorMessageControl)
         {
             // Render the spacing between the input and the error message
-            ComPtr<IAdaptiveErrorMessageConfig> errorMessageConfig;
-            RETURN_IF_FAILED(inputsConfig->get_ErrorMessage(errorMessageConfig.GetAddressOf()));
+            auto errorMessageConfig = inputsConfig.ErrorMessage();
 
-            ABI::AdaptiveCards::ObjectModel::Uwp::Spacing errorSpacing;
-            RETURN_IF_FAILED(errorMessageConfig->get_Spacing(&errorSpacing));
+            winrt::Spacing errorSpacing = errorMessageConfig.Spacing();
 
-            UINT spacing{};
-            RETURN_IF_FAILED(GetSpacingSizeFromSpacing(hostConfig.Get(), errorSpacing, &spacing));
-            auto separator = XamlHelpers::CreateSeparator(renderContext, spacing, 0, ABI::Windows::UI::Color());
+            uint32_t spacing = GetSpacingSizeFromSpacing(hostConfig, errorSpacing);
+
+            auto separator = XamlHelpers::CreateSeparator(renderContext, spacing, 0, winrt::Windows::UI::Color{});
+            auto inputValue = renderContext.GetInputValue(adaptiveInput);
 
-            ComPtr<IAdaptiveInputValue> inputValue;
-            RETURN_IF_FAILED(renderContext->GetInputValue(adaptiveInput, inputValue.GetAddressOf()));
             if (inputValue)
             {
-                RETURN_IF_FAILED(inputValue->put_ErrorMessage(errorMessageControl.Get()));
+                inputValue.ErrorMessage(errorMessageControl);
             }
-            XamlHelpers::AppendXamlElementToPanel(separator.Get(), stackPanelAsPanel.Get());
+
+            XamlHelpers::AppendXamlElementToPanel(separator, inputStackPanel);
 
             // Add the rendered error message
-            XamlHelpers::AppendXamlElementToPanel(errorMessageControl.Get(), stackPanelAsPanel.Get());
+            XamlHelpers::AppendXamlElementToPanel(errorMessageControl, inputStackPanel);
         }
 
-        // Create an AutomationPropertiesStatics object so we can set the accessibility properties that label allow us to use.
-        ComPtr<IAutomationPropertiesStatics> automationPropertiesStatics;
-        RETURN_IF_FAILED(
-            GetActivationFactory(HStringReference(RuntimeClass_Windows_UI_Xaml_Automation_AutomationProperties).Get(),
-                                 &automationPropertiesStatics));
-
+        // TODO: do I need this?
```

Comment:
@paulcam206 could you please have a look at this?

Reply:
don't think you need the dependency object -- the helper likely magically does the querying for you :)